saleor/graphql/account/mutations/service_account.py

```diff
@@ -0,0 +1,115 @@
+import graphene
+
+from saleor.core.permissions import get_permissions
+
+from ....account import models
+from ...core.enums import PermissionEnum
+from ...core.mutations import (
+    ClearMetaBaseMutation,
+    ModelDeleteMutation,
+    ModelMutation,
+    UpdateMetaBaseMutation,
+)
+
+
+class ServiceAccountInput(graphene.InputObjectType):
+    name = graphene.types.String(description="Name of the service account")
+    is_active = graphene.types.Boolean(
+        description="Determine if service account should be enabled"
+    )
+    permissions = graphene.List(
+        PermissionEnum,
+        description="List of permission code names to assign to this service account.",
+    )
+
+
+class ServiceAccountCreate(ModelMutation):
+    auth_token = graphene.types.String(
+        description="The newly created authentication token"
+    )
+
+    class Arguments:
+        input = ServiceAccountInput(
+            required=True,
+            description="Fields required to create a new service account.",
+        )
+
+    class Meta:
+        description = "Creates a new service account"
+        model = models.ServiceAccount
+        permissions = ("account.manage_service_account",)
+
+    @classmethod
+    def clean_input(cls, info, instance, data):
+        cleaned_input = super().clean_input(info, instance, data)
+        # clean and prepare permissions
+        if "permissions" in cleaned_input:
+            permissions = cleaned_input.pop("permissions")
+            cleaned_input["permissions"] = get_permissions(permissions)
+        return cleaned_input
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        instance = cls.get_instance(info, **data)
+        data = data.get("input")
```

Comment:
`data["input"]` instead as this field is required thus non-nullable?

_Interesting fact: apparently the cost difference is omissible (https://wiki.python.org/moin/PythonSpeed/PerformanceTips#Initializing_Dictionary_Elements)_

Reply:
I don't get it. We don't have such convention to use `['input']` when the field is required. 
Almost all of our mutations use `.get('input")` and I did the same to fit in our convention.

Reply:
I think both ways are fine; `data["input"]` may be more Pythonic if we now that `input` is always present, and this is the assumption for `ModelMutation`. If it's a problem, it's not here but in `ModelMutation` where this code was copied from. This mutation shouldn't override `perform_mutation` but `success_response` do assign the `auth_token` to the response.